cmd/clusterctl/client/tree/discovery_test.go

```diff
@@ -417,22 +428,29 @@ func Test_Discovery(t *testing.T) {
 				"cluster.x-k8s.io/v1beta1, Kind=Machine, ns1/cp1": {},
 				// Workers should have a machine deployment
 				"virtual.cluster.x-k8s.io/v1beta1, Kind=WorkerGroup, ns1/Workers": {
-					"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md1",
+					"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md01",
+					"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md02",
 				},
 				// Machine deployment should have a group of machines (grouping) and templates group
-				"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md1": {
+				"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md01": {
```

Comment:
Can you please add a similar check for `md02`.

Please group all md01 checks and md02 checks together. The test become more readable that way. :)